dask/array/tests/test_linalg.py

```diff
@@ -60,13 +62,53 @@ def test_tsqr_svd_irregular_blocks():
     data = from_array(mat, chunks=(3, n), name='A')[1:]
     mat2 = mat[1:, :]
 
-    u, s, v = tsqr(data, compute_svd=True)
+    u, s, vt = tsqr(data, compute_svd=True)
     u = np.array(u)
     s = np.array(s)
-    v = np.array(v)
+    vt = np.array(vt)
+    usvt = np.dot(u, np.dot(np.diag(s), vt))
+
+    s_exact = np.linalg.svd(mat2)[1]
 
-    assert np.allclose(mat2, np.dot(u, np.dot(np.diag(s), v)))  # accuracy check
+    assert np.allclose(mat2, usvt)  # accuracy check
     assert np.allclose(np.eye(n, n), np.dot(u.T, u))  # u must be orthonormal
-    assert np.allclose(np.eye(n, n), np.dot(v.T, v))  # v must be orthonormal
-    assert np.allclose(s, np.linalg.svd(mat2)[1])  # s must contain the singular
-                                                   # values
+    assert np.allclose(np.eye(n, n), np.dot(vt, vt.T))  # v must be orthonormal
+    assert np.allclose(s, s_exact)  # s must contain the singular values
+
+
+def test_svd_compressed():
+    m, n = 30, 25
+    r = 10
+    mat1 = np.random.randn(m, r)
+    mat2 = np.random.randn(r, n)
+    mat = mat1.dot(mat2)
+    data = from_array(mat, chunks=(10, 10), name='A')
+
+    n_iter = int(1e2)
+    for i in range(n_iter):
+        u, s, vt = svd_compressed(data, r)
+        u = np.array(u)
+        s = np.array(s)
+        vt = np.array(vt)
+        if i == 0:
+            usvt = np.dot(u, np.dot(np.diag(s), vt))
+        else:
+            usvt += np.dot(u, np.dot(np.diag(s), vt))
+    usvt /= n_iter
+
+    tol = 1e-1
+    assert np.allclose(np.linalg.norm(mat - usvt),
+                       np.linalg.norm(mat),
+                       rtol=tol, atol=tol)  # average accuracy check
```

Comment:
Ok, I will do that then.


On Sun, Jun 7, 2015 at 5:26 PM, Matthew Rocklin <notifications@github.com>
wrote:

> In dask/array/tests/test_linalg.py
> <https://github.com/ContinuumIO/dask/pull/280#discussion_r31880799>:
>
> > +    n_iter = int(1e2)
> > +    for i in range(n_iter):
> > +        u, s, vt = svd_compressed(data, r)
> > +        u = np.array(u)
> > +        s = np.array(s)
> > +        vt = np.array(vt)
> > +        if i == 0:
> > +            usvt = np.dot(u, np.dot(np.diag(s), vt))
> > +        else:
> > +            usvt += np.dot(u, np.dot(np.diag(s), vt))
> > +    usvt /= n_iter
> > +
> > +    tol = 1e-1
> > +    assert np.allclose(np.linalg.norm(mat - usvt),
> > +                       np.linalg.norm(mat),
> > +                       rtol=tol, atol=tol)  # average accuracy check
>
> I'm not asking for anything rigorous here, more like "Increasing q will
> increase accuracy but will also increase memory usage."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/280/files#r31880799>.
>